ts/test/session/sending/PendingMessageCache_test.ts

```diff
@@ -68,25 +67,66 @@ describe('PendingMessageCache', () => {
     expect(addedMessage.timestamp).to.deep.equal(rawMessage.timestamp);
   });
 
+  it('can add multiple messages belonging to the same user', async () => {
+    const device = TestUtils.generateFakePubKey();
+
+    await pendingMessageCacheStub.add(device, TestUtils.generateChatMessage());
+    // We have to timeout here otherwise it's processed too fast and messages start having the same timestamp
+    await timeout(5);
+    await pendingMessageCacheStub.add(device, TestUtils.generateChatMessage());
+    await timeout(5);
+    await pendingMessageCacheStub.add(device, TestUtils.generateChatMessage());
+
+    // Verify that the message is in the cache
+    const finalCache = await pendingMessageCacheStub.getAllPending();
+
+    expect(finalCache).to.have.length(3);
+  });
+
   it('can remove from cache', async () => {
     const device = TestUtils.generateFakePubKey();
     const message = TestUtils.generateChatMessage();
     const rawMessage = MessageUtils.toRawMessage(device, message);
 
     await pendingMessageCacheStub.add(device, message);
 
-    const initialCache = pendingMessageCacheStub.getAllPending();
+    const initialCache = await pendingMessageCacheStub.getAllPending();
     expect(initialCache).to.have.length(1);
 
     // Remove the message
     await pendingMessageCacheStub.remove(rawMessage);
 
-    const finalCache = pendingMessageCacheStub.getAllPending();
+    const finalCache = await pendingMessageCacheStub.getAllPending();
 
     // Verify that the message was removed
     expect(finalCache).to.have.length(0);
   });
 
+  it('should only remove messages with different timestamp and device', async () => {
+    const device = TestUtils.generateFakePubKey();
+    const message = TestUtils.generateChatMessage();
+    const rawMessage = MessageUtils.toRawMessage(device, message);
+
+    await pendingMessageCacheStub.add(device, message);
+    await timeout(5);
+    await pendingMessageCacheStub.add(
+      device,
+      TestUtils.generateChatMessage(message.identifier)
+    );
+    await pendingMessageCacheStub.add(TestUtils.generateFakePubKey(), message);
+
+    const initialCache = await pendingMessageCacheStub.getAllPending();
+    expect(initialCache).to.have.length(3);
+
+    // Remove the message
+    await pendingMessageCacheStub.remove(rawMessage);
+
+    const finalCache = await pendingMessageCacheStub.getAllPending();
+
+    // Verify that the message was removed
+    expect(finalCache).to.have.length(2);
```

Comment:
Does this ensure the intended message was removed?

Reply:
I'll add a test for that :)

Reply:
yeah might be good to something like this:
```suggestion
    expect(finalCache).to.have.length(2);
    const initialMessageFound = finalCache.find(
      m => m.device === message.device && m.timestamp === message.timestamp
    )
    expect(initialMessageFound).to.be.undefined;
```